From 983d1e36cbbc09bc03d0c6fb9ad22f29d748c8e0 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Sun, 26 Nov 2023 23:05:02 +0100
Subject: [PATCH 0074/1695] drivers: clk: assert registered clock pointer is
 not NULL

Fix clk_register() assertion entry instruction to print an assertion
error message on NULL clock pointer instead of issuing a segmentation
fault.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d2d69d1ab9303bf3f268276ecbb9ebf65b26c3dd drivers: clk: assert registered clock pointer is not NULL]
Change-Id: Ic334f93d4d2665ac225b9bbcc2ce9496efe279b4
Fixes: 2305544b3b9b ("drivers: clk: add generic clock framework")
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from bff6a8485115b8b36acc344dc83ef1cf440a8477]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/drivers/clk/clk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk.c b/core/drivers/clk/clk.c
index 6855b3d6e..d2ba19585 100644
--- a/core/drivers/clk/clk.c
+++ b/core/drivers/clk/clk.c
@@ -49,7 +49,7 @@ void clk_free(struct clk *clk)
 
 static bool __maybe_unused clk_check(struct clk *clk)
 {
-	if (!clk->ops)
+	if (!clk || !clk->ops)
 		return false;
 
 	if (clk->ops->set_parent && !clk->ops->get_parent)
-- 
2.47.2

